treebeard/forms.py

```diff
@@ -99,6 +97,9 @@ def __init__(self, data=None, files=None, auto_id='id_%s', prefix=None,
         # update the '_ref_node_id' choices
         choices = self.mk_dropdown_tree(opts.model, for_node=instance)
         self.declared_fields['_ref_node_id'].choices = choices
+        # use the formfield `to_python` method to coerse the field for custom ids
+        idFormField = opts.model._meta.get_field('id').formfield()
```

Comment:
This creates a dependency on the primary key being called `id`, so we need to change it to get the pk field regardless of its name.

Reply:
@rhunwicks True, but that dependency is actually quite a lot of places currently. I've created a PR on top of this one that allows for pk columns with alternative names. This might be a more invasive change though, so it may require a major version number update. See https://github.com/django-treebeard/django-treebeard/pull/193